pkg/virt-controller/watch/migration.go

```diff
@@ -85,16 +111,21 @@ func (md *MigrationDispatch) execute(store cache.Store, key string) error {
 		return setMigrationPhase(mig, kubev1.MigrationFailed)
 	}
 
-	obj, exists, err := store.GetByKey(key)
+	obj, exists, err := md.store.GetByKey(key)
 	if err != nil {
 		return err
 	}
 	if !exists {
 		return nil
 	}
 
-	var migration *kubev1.Migration = obj.(*kubev1.Migration)
-	logger := logging.DefaultLogger().Object(migration)
+	logger := logging.DefaultLogger().Object(obj.(*kubev1.Migration))
+	// Copy migration for future modifications
+	if obj, err = kubeapi.Scheme.Copy(obj.(runtime.Object)); err != nil {
+		logger.Error().Reason(err).Msg("could not copy migration object")
+		return err
+	}
+	migration := obj.(*kubev1.Migration)
 
```

Comment:
MigrationScheduled is removed. It is now neede anymore, now that everything is in one controller loop